src/uniprotkb/components/protein-data-views/ProteinNamesView.tsx

```diff
@@ -16,9 +16,34 @@ import {
 } from '../../adapters/namesAndTaxonomyConverter';
 import { ValueWithEvidence } from '../../types/modelTypes';
 
-export const NameWithEvidence = ({ data }: { data: ValueWithEvidence }) => (
+const ProteinName = ({
+  value,
+  withLink,
+}: {
+  value: string;
+  withLink: boolean;
+}) => {
+  const location = useLocation();
+  return (
+    <>
+      {withLink ? (
+        <Link to={`${location.pathname}#${stringToID(value)}`}>{value}</Link>
```

Comment:
There is a custom lint warning about not building the link path from scratch. I searched but couldn't find the necessary util.

Reply:
It's fine to ignore this if you're reusing an existing location object. The point was to avoid crafting URLs from scratch when I created this custom rule.
That being said, I did change a bit the code in order to reuse the location that the Link's `to` prop provides already instead of pulling it from the library's custom hook